### PR TITLE
Merge techx performance

### PR DESCRIPTION
**Pull request by crtrott**
This is a duplicate pull request in theory to #38. I merged that branch into develop and created a new one.

### CODE REVIEW

**Comment by srajama1** [comment]
@MicheldeMessieres : @crtrott says he has merged this in develop, but I don't see it. Do you know what happened ?

**Comment by MicheldeMessieres** [comment]
@srajama1 This branch was not added to develop. @crtrott fixed a PR update I had made which was not behaving as expected. He created this new branch to duplicate the original PR and then the issues were fixed, so this became the working PR branch.
I think the comments about merging with develop were just due to getting the branch fixed so it didn't show incorrect changes relative to develop.

**Comment by srajama1** [comment]
@MicheldeMessieres : Thanks for the clarification ! Let me take a look at the PR and see if we can merge.

**Comment by srajama1** [comment]
This has been pending for way too long due to my fault. I would like to merge it and add the SpGEMM benchmarks in our wiki as the tests for these. Can we check this is ready to be merged in the latest develop ?

File: perf_test/Kokkos_Performance_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+// ***********************************************************************
+// @HEADER
+
+#include "Kokkos_Performance.hpp"
+#include <Kokkos_Core.hpp>
+#include <iostream>
+#include <fstream>
+
+// For automatically determining hostname
+#if defined(_WIN32) || defined(__WIN32__) || defined(WIN32)
```
Comment (MicheldeMessieres) [inline perf_test/Kokkos_Performance_impl.hpp]: @srajama1 This windows code was something you had asked me to strip out but I didn't do yet. I could remove this on the next update.

**Comment by srajama1** [comment]
@MicheldeMessieres We are doing a release, do you have some cycles to clean this up ? We can push it in then.

**Comment by MicheldeMessieres** [comment]
@srajama1 For required changes I will the remove the old __WIN32__ reference and resolve the merge conflicts. I wasn't sure if this final version had been tried/reviewed so please let me know if there is anything else that should be changed.

**Comment by MicheldeMessieres** [comment]
Reconfigured this so will open a fresh PR.

**Pull request closed without merging**